### ISSUE TITLE
Lua script bug

### ISSUE THREAD

**Issue by yaoyi** [mhenrixon/sidekiq-unique-jobs#489]
current implementation
```lua
local cursor = "0"
repeat
  local pagination  = redis.call("SCAN", cursor, "MATCH", "*queue:*", "COUNT", count)
  local next_cursor = pagination[1]
  local queues      = pagination[2]
  for _, queue in ipairs(queues) do
    ...
    cursor = next_cursor
    ...
  end
until found == true or cursor == "0"
```
when next cursor is not 0,  the `queues` still could be empty, so should not put `cursor = next_cursor` inside the for loop which might let cursor have no chance to get updated

**Comment by mhenrixon** [comment]
Thanks for the report @yaoyi, I’ll have a look at this today.

**Comment by mhenrixon** [comment]
Released as v7.0.15
